src/promptflow/tests/executor/e2etests/test_batch_engine.py

```diff
@@ -162,50 +155,61 @@ def test_batch_with_metrics(self, dev_connections):
         assert isinstance(batch_results.metrics, dict)
         assert batch_results.metrics == get_flow_expected_metrics(flow_folder)
-        status_summary = batch_results.get_status_summary()
-        assert status_summary == get_flow_expected_status_summary(flow_folder)
+        assert batch_results.total_lines == batch_results.completed_lines
+        assert batch_results.node_status == get_flow_expected_status_summary(flow_folder)
 
     def test_batch_with_partial_failure(self, dev_connections):
         flow_folder = SAMPLE_FLOW_WITH_PARTIAL_FAILURE
         inputs_mapping = {"idx": "${data.idx}", "mod": "${data.mod}", "mod_2": "${data.mod_2}"}
         batch_results = submit_batch_run(flow_folder, inputs_mapping, connections=dev_connections)
         assert isinstance(batch_results, BatchResult)
-        status_summary = batch_results.get_status_summary()
-        assert status_summary == get_flow_expected_status_summary(flow_folder)
+        assert batch_results.total_lines == 10
+        assert batch_results.completed_lines == 5
+        assert batch_results.failed_lines == 5
+        assert batch_results.node_status == get_flow_expected_status_summary(flow_folder)
 
     def test_batch_with_line_number(self, dev_connections):
         flow_folder = SAMPLE_FLOW_WITH_PARTIAL_FAILURE
         input_dirs = {"data": "inputs/data.jsonl", "output": "inputs/output.jsonl"}
         inputs_mapping = {"idx": "${output.idx}", "mod": "${data.mod}", "mod_2": "${data.mod_2}"}
-        batch_results = submit_batch_run(
-            flow_folder, inputs_mapping, input_dirs=input_dirs, connections=dev_connections
+        batch_results, output_dir = submit_batch_run(
+            flow_folder, inputs_mapping, input_dirs=input_dirs, connections=dev_connections, return_output_dir=True
         )
         assert isinstance(batch_results, BatchResult)
-        assert len(batch_results.outputs) == 2
-        assert batch_results.outputs == [
+        outputs = load_jsonl(output_dir / OUTPUT_FILE_NAME)
+        assert len(outputs) == 2
+        assert outputs == [
             {"line_number": 0, "output": 1},
             {"line_number": 6, "output": 7},
         ]
 
     def test_batch_with_openai_metrics(self, dev_connections):
         inputs_mapping = {"url": "${data.url}"}
-        batch_result = submit_batch_run(SAMPLE_FLOW, inputs_mapping, connections=dev_connections)
+        batch_result, output_dir = submit_batch_run(
+            SAMPLE_FLOW, inputs_mapping, connections=dev_connections, return_output_dir=True
+        )
         nlines = get_batch_inputs_line(SAMPLE_FLOW)
-        assert len(batch_result.outputs) == nlines
-        openai_metrics = batch_result.get_openai_metrics()
-        assert "total_tokens" in openai_metrics
-        assert openai_metrics["total_tokens"] > 0
+        outputs = load_jsonl(output_dir / OUTPUT_FILE_NAME)
+        assert len(outputs) == nlines
+        assert batch_result.system_metrics.total_tokens > 0
```

Comment:
Add assert for `prompt_tokens` and `completion_tokens`?

Reply:
Updated.